plugins/lookup/binary_file.py

```diff
@@ -0,0 +1,115 @@
+options:
+  _terms:
+    description:
+      - Paths of the files to read.
+      - Relative paths will be searched for in different places. See R(Ansible task paths, playbook_task_paths) for more details.
+    required: true
+    type: list
+    elements: str
+  not_exist:
+    description:
+      - Determine how to react if the specified file cannot be found.
+    type: str
+    choices:
+      error: Raise an error.
+      empty: Return an empty string for the file.
+      empty_str:
+        - Return the string C(empty) for the file.
+        - This cannot be confused with Base64 encoding due to the missing padding.
+    default: error
+notes:
+  - This lookup does not understand 'globbing' - use the P(ansible.builtin.fileglob#lookup) lookup instead.
+seealso:
+  - plugin: ansible.builtin.b64decode
+    plugin_type: filter
+    description:
+      - The b64decode filter can be used to decode Base64 encoded data.
+      - Note that Ansible cannot handle binary data, the data will be interpreted as UTF-8 text!
+  - plugin: ansible.builtin.file
+    plugin_type: lookup
+    description: You can use this lookup plugin to read text files from the Ansible controller.
+  - module: ansible.builtin.slurp
+    description:
+      - Also allows to read binary files Base64 encoded, but from remote targets.
+      - >-
+        With C(delegate_to: localhost) can be redirected to run on the controller, but you have to know the path to the file to read.
+        Both this plugin and P(ansible.builtin.file#lookup) use some search path logic to for example also find files in the C(files)
+        directory of a role.
+  - ref: playbook_task_paths
+    description: Search paths used for relative files.
+"""
+
+EXAMPLES = r"""
+---
+- name: Output Base64 contents of binary files on screen
+  ansible.builtin.debug:
+    msg: "Content: {{ lookup('community.general.binary_file', item) }}"
+  loop:
+    - some-binary-file.bin
+"""
+
+RETURN = r"""
+_raw:
+  description:
+    - Base64 encoded content of requested files, or an empty string resp. the string C(empty), depending on the O(not_exist) option.
+    - This list contains one string per element of O(_terms) in the same order as O(_terms).
+  type: list
+  elements: str
+  return: success
+"""
+
+import base64
+
+from ansible.errors import AnsibleLookupError
+from ansible.plugins.lookup import LookupBase
+
+from ansible.utils.display import Display
+
+display = Display()
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
+        self.set_options(var_options=variables, direct=kwargs)
+        not_exist = self.get_option("not_exist")
+
+        result = []
+        for term in terms:
+            display.debug(f"Searching for binary file: {term!r}")
+            path = self.find_file_in_search_path(variables, "files", term, ignore_missing=(not_exist != "error"))
```

Comment:
The implementation of `find_file_in_search_path` can be found here: https://github.com/ansible/ansible/blob/4c04b8c7c3f43b9c50831e97e4479e428cef9bd6/lib/ansible/plugins/lookup/__init__.py#L123-L140